ui/scripts/system.js

```diff
@@ -29,9 +29,19 @@
     var mapRouterType = function (index, router) {
         var routerType = _l('label.menu.system');
 
-        if (router.projectid) routerType = _l('label.project');
-        if (router.vpcid) routerType = _l('label.vpc');
-        if ("isredundantrouter" in router && router.isredundantrouter) routerType = routerType + " (" + router.redundantstate + ")";
+        if (router.projectid) {
+            routerType = _l('label.project');
+            router.account = router.project;
+        }
+
+        if (router.vpcid) {
+            routerType = _l('label.vpc');
+            router.guestnetworkname = router.vpcname;
+        }
+
+        if ("isredundantrouter" in router && router.isredundantrouter) {
```

Comment:
I believe you can use directly `router.isredundantrouter` here without needing to check if the property `isredundantrouter` exists in the object `router`; `undefined `is interpreted as `false`.

Reply:
Thanks, I get defensive writing code. I guess this is not be necessary here.